### PR TITLE
Fix `pyramidindices` performance

### LINKED ISSUE #14: pyramideindices is extremely inefficient
I made the mistake to use it in a performance critical part of the code, and noticed that `pyramideindices` re-eval its body at every execution, making it very inefficient:
```julia
julia> @time pyramidindices(3, 3) ;
  0.012201 seconds (23.48 k allocations: 1.190 MiB, 88.89% compilation time)

julia> @time pyramidindices(3, 3) ;
  0.012611 seconds (23.48 k allocations: 1.190 MiB, 86.13% compilation time)  # Compilation again
```
I am not quite sure how to solve the issue in general.

For my case I just cache the result with Memoization.jl and it is fine (I call it with the same arguments over and over, so I don't need to be very fancy).

### PR DESCRIPTION
**Pull request by lpawela**
This fixes #14 using `@generated` functions.

### CODE REVIEW

File: pi.jl [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+dims, tensize = ndims(d), size(d, 1)
+@time pi2(dims, tensize)
+@time pi2(dims, tensize)
+
+@time e = rand(20, 20, 20)
+dims, tensize = ndims(e), size(e, 1)
+@time pi2(dims, tensize)
+@time pi2(dims, tensize)
+
+
```
Comment (kdomino) [inline pi.jl]: Do we need the ```pi.jl``` file at all? 
By the way it does not compile while running ```julia> pi.jl```.

**Review by kdomino — CHANGES_REQUESTED** [review]
It is fast, and more efficient, I think. Just nor sure about ```pi.jl``` file

File: pi.jl (hunk 1 above)
Comment (pgawron) [reply pi.jl]: Shouldn't it be in the `test directory`?

**Review by pgawron — CHANGES_REQUESTED** [review]
The tests have to be fixed. `pi.jl` file is in wrong place.

File: pi.jl (hunk 1 above)
Comment (lpawela) [reply pi.jl]: File removed, it was for some timing comparison and is not needed.

**Review by kdomino — APPROVED** [review]

**Pull request merged**